### PR TITLE
Docs: Clean up Roadmap and What's New page

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Remove SideNav from Roadmap and What's New pages, fix Footer alignment

#### Why?

Remove redundant side nav from these standalone pages. also adds a Summary field for weekly/monthly digest posts

Future Updates: Centering the content on these pages 

Screen Shot 2022-09-06 at 11 15 44 AM

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4809)
- [Figma](https://www.figma.com/file/emykswARRS1tQwpB7ThIPN/Gestalt-2022-Site?node-id=998%3A19503)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [X] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/docs-components/App.js [hunk 1]
```diff
@@ -17,7 +17,6 @@ type Props = {|
 
 export default function App({ children, files }: Props): Node {
   const router = useRouter();
-  const [isHomePage, setIsHomePage] = useState(router?.route === '/home');
```
Comment (ayeshakmaz) [inline docs/docs-components/App.js:20]: Moved all this logic to App.js

File: docs/docs-components/Header.js [hunk 2]
```diff
@@ -173,7 +172,7 @@ function Header() {
           <Text color="default" weight="bold">
             <Link
               accessibilityLabel="Gestalt home"
-              href="/"
+              href="/home"
```
Comment (ayeshakmaz) [inline docs/docs-components/Header.js:175]: Prevents the flashing side nav when clicking the logo

File: docs/redirects.js [hunk 3]
```diff
@@ -162,8 +162,8 @@ const misc = [
     permanent: true,
   },
   {
-    source: '/whats_new',
-    destination: '/roadmap/whats_new',
+    source: '/roadmap/whats_new',
+    destination: '/whats_new',
```
Comment (ayeshakmaz) [inline docs/redirects.js:166]: Switching these back, in case people made bookmarks already

File: docs/docs-components/AppLayout.js [hunk 4]
```diff
@@ -15,16 +16,26 @@ const CONTENT_MAX_WIDTH_PX = 1546;
...
+  const router = useRouter();
+
+  const [isHomePage, setIsHomePage] = useState(router?.route === '/home');
+  const [shouldHideSideNav, setShouldHideSideNav] = useState(true);
 
   const footerColor =
     colorScheme === 'dark' ? 'var(--color-gray-roboflow-700)' : 'var(--color-orange-firetini-0)';
 
+  useEffect(() => {
+    const fullWidthPages = ['home', 'whats_new', 'roadmap'];
```
Comment (dangerismycat) [inline docs/docs-components/AppLayout.js:33]: nit: this should be a top-of-file const

File: docs/docs-components/AppLayout.js [hunk 5]
```diff
@@ -15,16 +16,26 @@ const CONTENT_MAX_WIDTH_PX = 1546;
...
-  isHomePage?: boolean,
 |};
 
-export default function AppLayout({ children, colorScheme, isHomePage }: Props): Node {
+export default function AppLayout({ children, colorScheme }: Props): Node {
   const { isMobile } = useDocsDeviceType();
   const { isSidebarOpen, setIsSidebarOpen } = useNavigationContext();
+  const router = useRouter();
+
+  const [isHomePage, setIsHomePage] = useState(router?.route === '/home');
```
Comment (dangerismycat) [inline docs/docs-components/AppLayout.js:26]: Does this need to be a state variable? It seems like it could be a simple check (`router?.route === '/home'`) in the main body of the function component. I'm also confused why we're referencing `router.route` here and `window.location.pathname` below

File: docs/docs-components/AppLayout.js [hunk 6]
```diff
@@ -85,29 +96,28 @@ export default function AppLayout({ children, colorScheme, isHomePage }: Props):
...
-            >
-              <DocsSideNavigation showBorder />
-            </Box>
+        <Box minWidth={MIN_NAV_WIDTH_PX} mdDisplay={shouldHideSideNav ? 'none' : 'block'}>
+          <Box
+            display="none"
+            mdDisplay="block"
+            position="fixed"
+            overflow="auto"
+            height="calc(100% - 75px)"
```
Comment (dangerismycat) [inline docs/docs-components/AppLayout.js:105]: Where does this 75px come from? Seems a bit magical

File: docs/redirects.js (hunk 3 above)
Comment (dangerismycat) [reply docs/redirects.js:166]: _sigh_ this is why I wanted the flattest URL structure possible…hopefully we don't accumulate too many more of these in the future

File: docs/pages/BlogPosts.json [hunk 7]
```diff
@@ -3,6 +3,7 @@
   "digests": [
     {
       "week": "September 02, 2022",
+      "summary": "Handoff kit updates, Figma search improvements, and more mobile docs",
```
Comment (dangerismycat) [inline docs/pages/BlogPosts.json:6]: Can we split these changes out to a separate PR? It's a pretty different change and we probably want to iterate on it a bit

**Review by dangerismycat — APPROVED** [review]
